x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/session/VersionCompatibilityChecks.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.sql.session;
+
+import org.elasticsearch.xpack.ql.type.DataType;
+import org.elasticsearch.xpack.ql.type.DataTypes;
+import org.elasticsearch.xpack.sql.proto.SqlVersion;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.elasticsearch.Version.V_7_13_0;
+import static org.elasticsearch.xpack.ql.type.DataTypes.isArray;
+import static org.elasticsearch.xpack.sql.type.SqlDataTypes.types;
+
+public final class VersionCompatibilityChecks {
+
+    public static final SqlVersion INTRODUCING_ARRAY_TYPES = SqlVersion.fromId(V_7_13_0.id);
```

Comment:
I wonder if we need to remove `INTRODUCING_` since it will be the same for all new features. Then again, we need to differentiate from those that are being removed.